colcon_ros/task/catkin/build.py

```diff
@@ -73,6 +73,19 @@ def add_arguments(self, *, parser):  # noqa: D102
                         'python2path', Path(args.install_base),
                         self.context.pkg.name,
                         'PYTHONPATH', str(rel_python_path), mode='prepend')
+
+        shell_extensions = get_shell_extensions()
+        file_extensions = []
+        for shell_extensions_same_prio in shell_extensions.values():
+            for shell_extension in shell_extensions_same_prio.values():
+                file_extensions += shell_extension.get_file_extensions()
+
+        # register custom hooks created via catkin_add_env_hooks
+        custom_hooks_path = Path(args.install_base) / \
+            'etc' / 'catkin' / 'profile.d'
+        for file_extension in sorted(file_extensions):
```

Comment:
Sorting the file extensions might be a problem if e.g. a `bash` script is sourced before a `sh` script but it requires "stuff" from that `sh` script. Therefore please keep the shell extension order defined by their priorities.

Reply:
I think I understand - I was mirroring the behaviour in https://github.com/colcon/colcon-ros/blob/master/colcon_ros/task/ament_cmake/build.py#L48 which may not be correct here.

I've left file_extensions unsorted, however I realized that the globbed custom hooks _should_ probably be sorted by file name.